.github/workflows/integration-tests.yaml

```
@@ -13,29 +13,37 @@ permissions:
+            source:
+              - '**/*'
+              - '!docs/**'
               - '!.spelling'
               - '!README.md'
-              - '!docs/**'
+              - '!**/*.md'
+              - '!.github/**/*'
+
+  integration-tests:
+    needs: changes
+    if: needs.filter.outputs.source-changed == 'true'
```

actionlint (1.7.4)

property "filter" is not defined in object type {changes: {outputs: {source-changed: string}; result: string}}

(expression)